modules/tensor_mechanics/src/kernels/lagrangian/LagrangianStressDivergenceBase.C

```diff
@@ -0,0 +1,121 @@
+#include "LagrangianStressDivergenceBase.h"
+
+InputParameters
+LagrangianStressDivergenceBase::validParams()
+{
+  InputParameters params = Kernel::validParams();
+
+  params.addRequiredParam<unsigned int>("component", "Which direction this kernel acts in");
+  params.addRequiredCoupledVar("displacements", "The displacement components");
+
+  params.addParam<bool>("large_kinematics", false, "Use large displacement kinematics");
+  params.addParam<bool>("stabilize_strain", false, "Average the volumetric strains");
+
+  params.addParam<std::string>("base_name", "Material property base name");
+
+  params.addRequiredRangeCheckedParam<unsigned int>("component",
+                                                    "component < 3",
+                                                    "An integer corresponding to the direction "
+                                                    "the variable this kernel acts in. (0 for x, "
+                                                    "1 for y, 2 for z)");
+
+  return params;
+}
+
+LagrangianStressDivergenceBase::LagrangianStressDivergenceBase(const InputParameters & parameters)
+  : Kernel(parameters),
+    _large_kinematics(getParam<bool>("large_kinematics")),
+    _stabilize_strain(getParam<bool>("stabilize_strain")),
```

Comment:
It will still work if you use it for the "wrong" element types, it's just not going to do what you want.  Given that I added a check in the action, rather than in the kernels.  So a determined user could still enable it if they wanted to but most people who use the action will be warned off.